botocore/endpoint.py

```diff
@@ -75,7 +77,26 @@ def convert_to_response_dict(http_response, operation_model):
 
 
 class PreserveAuthSession(Session):
+    """Internal session class used to workaround requests behavior.
+
+    This class is intended to be used only by the Endpoint class.
```

Comment:
I think the class should be renamed. It is no longer just for preventing the rebuilding of auth. It also is used to customize the adapter and the connection pool. Maybe call it ``BotocoreRequestsSession``?